docs/rendering_system.rst

```diff
@@ -0,0 +1,241 @@
+..
+   Author: Adam Turner
+
+   We can't use :pep:`X` references in this document, as they are relative from
+   the content root.
+   We use :doc:`PEP X <../pep-xxxx>` instead.
+
+
+An Overview of the PEP Rendering System
+=======================================
+
+This document provides an overview of the PEP rendering system, as a companion
+to :doc:`PEP 676 <../pep-0676>`.
+
+
+1. Configuration
+----------------
+
+Configuration is held in three files:
+
+- ``conf.py`` contains the majority of the configuration
+- ``contents.rst`` contains the table of contents directives for Sphinx
+- ``pep_sphinx_extensions/pep_theme/theme.conf`` sets the Pygments themes
+
+The configuration:
+
+- registers the custom Sphinx extension
+- sets both ``.txt`` and ``.rst`` suffixes to be parsed as PEPs
+- tells Sphinx which source files to use
+- registers the PEP theme, maths renderer, and template
+- disables some default settings that are covered in the extension
+- sets the default and 'dark mode' code formatter styles
+
+
+2. Orchestration
+----------------
+
+``build.py`` acts to manage the rendering process.
+Usage is covered in :doc:`build`.
+
+
+3. Extension
+------------
+
+The Sphinx extension and theme are contained in the ``pep_sphinx_extensions``
+directory.
+The following is a brief overview of the stages of the PEP rendering process,
+and how the extension functions at each point.
+
+
+3.1 Extension setup
+~~~~~~~~~~~~~~~~~~~
+
+The extension registers several objects:
+
+- ``FileBuilder`` and ``DirectoryBuilder`` run the build process for file- and
+  directory- based building
+- ``PEPParser`` registers the custom document transforms and parses PEPs to
+  a Docutils document.
+- ``PEPTranslator`` converts a Docutils document into HTML
+- ``PEPRole`` handles ``:pep:`` roles in the reStructuredText source
+
+It also patches default behaviour:
+
+- updating the default settings
+- updating the Docutils inliner
+- using HTML maths display over MathJax
+
+
+3.2 Builder initialised
+~~~~~~~~~~~~~~~~~~~~~~~
+
+After the Sphinx builder object is created and initialised, we ensure the
+configuration is correct for the builder chosen.
+
+Currently this involves updating the relative link template.
+See ``_update_config_for_builder`` in ``pep_sphinx_extensions/__init__.py``.
+
+
+3.3 Before documents are read
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``create_pep_zero`` hook is called. See `4. PEP 0 generation`_.
+
+
+3.4 Read document
+~~~~~~~~~~~~~~~~~
+
+Parsing the document is handled by ``PEPParser``
+(``pep_sphinx_extensions.pep_processor.parsing.pep_parser.PEPParser``), a light
+wrapper over ``sphinx.parsers.RSTParser``.
```

Comment:
```suggestion
(``pep_sphinx_extensions.pep_processor.parsing.pep_parser.PEPParser``),
a lightweight wrapper over ``sphinx.parsers.RSTParser``.
```

It took me a second to parse this, since "lightweight" is the more idiomatic and conventionally-used word in this specific context